### DISCUSSION TITLE
How to get something like remark-attr or rehype-attr to add props to md components like lists?

### DISCUSSION THREAD

**Discussion by schlichtanders** [mdx-js/mdx#2144] [Q&A] [answered]
background: I am combining mdx with astro and reveal.js and would now like to have a way to add props to the markdown components. One usecase is adding the class `fragment` to an arbitrary part, or adding extra props to the `<code>` part of the code-fences. 

Importantly, these are attributes which are individual to a single list/code-block, etc., so nothing which I would like to define for all lists, all code-blocks, etc., but really something which should be defined locally.

I found [remark-attr](https://www.npmjs.com/package/remark-attr) and [rehype-attr](https://www.npmjs.com/package/rehype-attr) but it seems both aren't supported with mdx, because mdx blocks both the curly syntax from remark-attr as well as the html-comment syntax from rehype-attr.

Is there another way to add props (attributes) to markdown components in mdx?

**Comment by wooorm** [comment, marked as answer]
Use markdown for the common things:

```markdown
`code`
```

Use JSX for the exceptions:

```markdown
<code className="special">code</code>
```

**Reply by schlichtanders** [reply]
Thank you very much for your help

okay, so there really is no common way to get something inbetween?
- I've only seen specific support for pre/code blocks in https://github.com/remcohaszing/remark-mdx-code-meta

If there would be a way to insert HTML-Comments so that rehype-attr sees them, that would be nice, but I guess this does not exist right?

**Reply by wooorm** [reply]
Correct, the `meta` attribute is the only place (other than elements) to “stuff” arbitrary data, that is otherwise hidden.

You can’t use HTML comments in MDX, because it doesn’t support HTML. You can use JavaScript instead (in braces), which means that you can use JavaScript comments in braces.

**Reply by schlichtanders** [reply]
yeah, got the comment thing...
just experimented a bit by bringing the comment into its own component (using Astro component), but also this does not get picked up by rehype-attr, probably because it is inside a component which is hidden for rehype plugins...

Let me know if you come across some `mdx-attr` kind of thingy

**Reply by schlichtanders** [reply]
@wooorm is the a documentation about the meta attribute and where it is supported?

I mean it seems to be supported for code tags, is it also supported for lists? how?

**Reply by wooorm** [reply]
sure, makes sense that `rehype-attr` does not understand MDX? If you want that, best to ask them!

for `meta`, I am not sure what you mean. It’s markdown. There’s the CommonMark spec if that’s what you mean? Maybe https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field is of help?

**Reply by schlichtanders** [reply]
thank you for the pointer, so it seems that such meta data is only supported at code elements

**Reply by schlichtanders** [reply]
the custom rehype plugin  example in the doc https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field 
which puts the meta attribute to props does not work for me unfortunately

**Reply by schlichtanders** [reply]
thank you very much for your help!
going offline now
